### ISSUE TITLE
0.0 - Prework - Michael Esposito

### ISSUE THREAD

**Issue by Esponage** [TIY-GVL-FEE-2015-August/assignments#3]
# Prework

## Hello and welcome to Front End Engineering!

I hope you're excited about this new possibility, this turning point in your career. I am. I'm really looking forward to meeting each of you in person and getting to know you better over the course of the upcoming 12 weeks. I'm also really excited about teaching and mentoring you all and seeing what awesome things you'll build and what interesting ways you'll approach the assignments and projects in the class.

To get started, we'll be using [Github](http://github.com) extensively. You need to [sign up for a free Github account](http://github.com/signup), find [this repo](https://github.com/TIY-GVL-FEE-2015-August/assignments) and "star" it by clicking on the "Star" button, so that I know who to give access to. Once you're added, follow the steps below, which are _due the first day of class_.

Throughout the course, I'll be adding files to this repository to distribute assignments and to [the notes page](http://jacobthemyth.gitbooks.io/tiy-gvl-fee-2015-august/content/) with notes for each class and general information about the course. Make sure to bookmark both of these pages since you will be visiting them on a daily basis. If you have any questions about any of the content, please [open an Issue on the `questions` repository](https://github.com/TIY-GVL-FEE-2015-August/questions/issues), and I'll attempt to answer you promptly.

## Prework Instructions

There are quite a few tasks and learning topics that are necessary to take care of before we can actually start learning how to be front end developers, so these tasks are meant to help you some of the busy work out of the way before class starts.

* Get Started
    * [x] [Create a Github account](https://github.com/signup)
    * [x] [Star the assignment repo by clicking on the "Star" button](https://github.com/TIY-GVL-FEE-2015-August/assignments)  
      - [More info about stars](https://help.github.com/articles/about-stars)

* Create your assignment issue
    * [x] [Create a new Issue in the class repo](https://github.com/TIY-GVL-FEE-2015-August/assignments/issues)
    * [x] Give it a title according to the assignment name with your name appended, e.g. `0.0 - Prework - Jacob Smith`
    * [x] Paste the _raw_ contents of this `README.md` file as the "description"
      - read [Mastering Issues](https://guides.github.com/features/issues/) for more information.
      - **Hint**: You will need to click on the `README.md` file in the file explorer at the top of the page, then look for the "Raw" button to copy the raw text. After pasting the text in a new issue, you should end up with a bunch of checkboxes just like those below, thanks to [the way that Github interprets Markdown](https://guides.github.com/features/mastering-markdown/).

    As you complete tasks, check off the corresponding checkboxes. This is your **Work In Progress (WIP) Issue**. You'll create a new one for each Assignment to show your work and track your progress.

* [Edit your Github profile](https://github.com/settings/profile) to provide:
    * [x] A picture, so I can recognize your face.
    * [x] Your full name, so I can recognize your name.
    * [x] A valid public email address, so I can contact you if I need to.

* Install Atom.app.

  Atom is a text editor made specifically for writing code, so we'll be using it a lot in class.

  * [x] [Download Atom](https://atom.io/download/mac)
  * [x] Unzip the file and drag it to your /Applications directory.
  * [x] Drag it to your Dock so you have quick access

* Learn some HTML and CSS

  HTML and CSS are the foundation of all the content of a webpage or web app.  However, we will only be spending about a week on HTML and CSS, so you should have a basic foundation before the class starts.

  The linked resource walks you through the basics. Anywhere it references a "text editor", you should use Atom.app. Also, make sure not to copy and paste the examples, but actually type out all of the HTML and CSS.

  * [x] Complete [Building Your First Web Page](http://learn.shayhowe.com/html-css/building-your-first-web-page/)
  * [x] Complete [Getting to Know HTML](http://learn.shayhowe.com/html-css/getting-to-know-html/)
  * [x] Read through [Getting to Know CSS](http://learn.shayhowe.com/html-css/getting-to-know-css/)
  * [x] Complete [Opening the Box Model](http://learn.shayhowe.com/html-css/opening-the-box-model/)
  * (Optional) Complete the rest of the lessons on learn.shayhowe.com

* Make your first repository on GitHub
  * [x] [Complete the "Hello World" Github guide](https://guides.github.com/activities/hello-world/)
  * [x] Add a link to your new Github repo as a comment in your WIP Issue.

* Push your code from learning HTML and CSS to GitHub
  * [x] Complete [Getting your project on GitHub](https://guides.github.com/introduction/getting-your-project-on-github/) to publish your webpage from above.
  * [x] Add a link to your new Github repo as a comment in your WIP Issue.

* [x] Close this issue! You're done!

## Extracurricular Activities

These aren't required per se, but might be helpful if you have time available.

### Typing

Believe it or not, *a lot* of time in the first weeks of programming is spent fixing simple typing errors. It would not be ridiculous to suggest that many of you will spend 12-24 hours (or more) fixing typing errors during class.

If you don't know how to touch type (i.e. type without looking at the keyboard), try to spend some time each day working on touch typing (you can try http://www.keybr.com/). If you already know how to touch type, I would highly recommend working through the JavaScript lessons on http://typing.io. Typing for programming is actually a good bit different than everyday typing, so getting some practice in could potentially save you a ton of time the first few weeks of class.

### JavaScript

JavaScript is the programming language that allows web pages to react and change as we use them, instead of waiting on a page load to change content. It will form the backbone (no pun intended) of what we learn during the front end course, so it's important to have a general idea of how it works before we start. I recommended that you go through this course during our interviews, so if you've already completed it, feel free to move on to the next section.

Complete the Codecademy course at http://www.codecademy.com/tracks/javascript. This should take you approximately 10 hours, depending on your prior experience.

### The Command Line

The command line is an interface to the internals of your computer, as opposed to the graphical user interface (GUI) that we commonly use. We will be using the command line extensively in our class, but the point of this prework is not to be an expert, just to start to get comfortable.

Especially if you've never used the command line before, you may want to go through these more than once. Make sure not just to read/watch, but really dive in and use the commands that you learn about.

1. Read http://skillcrush.com/2012/12/03/command-line-2/
2. Go through the Treehouse course on [Getting Started with the Console](http://teamtreehouse.com/library/console-foundations#getting-started-with-the-console)
3. Go through the Treehouse course on [Users and Permissions](http://teamtreehouse.com/library/console-foundations#users-and-permissions)
4. Go through the Treehouse course on [Environment and Redirection](http://teamtreehouse.com/library/programming/console-foundations#environment-and-redirection)
5. Read and practice with [A Command Line Primer for Beginners](http://lifehacker.com/5633909/who-needs-a-mouse-learn-to-use-the-command-line-for-almost-anything)

### git

Git is a free and open source distributed version control system designed to handle everything from small to very large projects with speed and efficiency.

It is a command line tool that we will be using for every project we do in class. Again, the objective is not to become an expert before the class, but just to start to get comfortable.

Go through the Treehouse course on [Git Basics](http://teamtreehouse.com/library/git-basics)

If you want to spend some more time with git, [Git immersion](http://gitimmersion.com/) is a great resource.

# Learning

Finally, if you're interested in learning how best to study and learn, many of the principles that I use to design our course can be found in the book [Make it Stick](http://www.amazon.com/Make-It-Stick-Successful-Learning/dp/0674729013)([public library](http://www.worldcat.org/title/make-it-stick-the-science-of-successful-learning/oclc/859168651&referer=brief_results)).

**Comment by Esponage** [comment]
https://github.com/Esponage/hello-world

**Comment by Esponage** [comment]
<b>Styles Conference Assignment</b>

https://github.com/Esponage/Styles-Conference.git

**Comment by Esponage** [comment]
<b>Styles Conference Assignment</b>

https://github.com/Esponage/Styles-Conference.git
